plugins/SegmentEditor/API.php

```diff
@@ -348,6 +349,52 @@ public function add(
         return $id;
     }
 
+    /**
+     * Stars a stored segment.
+     *
+     * @param int $idSegment
+     * @throws Exception if the user is not logged in or does not have the required permissions.
+     */
+    public function star(int $idSegment): ?array
```

Comment:
When can this return null? If it throws an exception, it's not considered as return null, otherwise always an array.

```suggestion
    public function star(int $idSegment): array
```

We should also add a PHPStan compatible return type description on the response structure and the expected array keys.

Reply:
✅ Done!